src/naming.jl

```diff
@@ -86,12 +89,13 @@ function savename(prefix::String, c, suffix::String;
         prefix = joinpath(prefix, dpre)
     end
 
-    labels = vecstring(accesses) # make it vector of strings
+    visibles = setdiff(accesses, ignores)
+    labels = vecstring(visibles) # make it vector of strings
```

Comment:
Currently `allaccess` will return `Vector{String}` or `Vector{Symbol}`. So to make this work, `accesses` and `ignores` must be of the same type. If you switch the lines and convert both with `vecstring` before doing `setdiff` the types don't matter because everything is a string afterwards.

Reply:
Good point! I will fix this.

Reply:
It actually introduces some issue later in the code. Basically the current way of implement the sort is by returning the indics of `visibles` (previously `accesses`) to `p` and latter access it by `visibles[i]`(previously `accesses[i]`). This is necessary because we need the original variables in `visibles` (or `accesses`) for `access`. However, this relies on the fact that the label and value in the label and original container has consistent indics, which is not the case after `setdiff`.

I have to change the way labels and original values are accessed using `zip` and `filter` - please see my updates.

Reply:
@Datseris Please see here on why simply swapping the lines doesnt work.

Reply:
It might be hard to understand it by words. Here is what would happen if I simply change the two lines.

```julia
    labels_acc, labels_ign = vecstring(accesses), vecstring(ignores)  # L1
    labels = setdiff(labels_acc, labels_ign)                          # L2
    p = sortperm(labels)                                              # L3
    first = prefix == "" || endswith(prefix, PATH_SEPARATOR)          # L4
    s = prefix                                                        # L5
    for j ∈ p                                                         # L6
        val = access(c, accesses[j])                                  # L7
        label = labels[j]                                             # L8
```

Given `accesses = [:z, :y, :x]` and `ignores = [:z]`
- L1: `labels_acc <- ["z", "y", "x"]` and `labels_ign <- ["z"]`
- L2: `labels <- ["y", "x"]`
- L3: `p <- [2, 1]`
- Loop (first time)
  - L6: `j <- 2`
  - L7: `accesses[j]` is `:y`
  - L8: `labels[j]` is `"x"`
- Loop (second time)
  - L6: `j <- 1`
  - L7: `accesses[j]` is `:z`
  - L8: `labels[j]` is `"y"`

As you can see, `accesses[j]` is not consistent with `labels[j]`.

Reply:
Yes, I have understood that, which is why in my latest comment I suggest the `continue` version. When I say "original" version of the code, I literally mean the version before your PR, here: https://github.com/JuliaDynamics/DrWatson.jl/blob/master/src/naming.jl#L89-L94

Adding a `labels[j] in ignore_labels && continue` would fix everything. See https://github.com/JuliaDynamics/DrWatson.jl/pull/123#issuecomment-583779631 . Please notice what I suggest: there is no `labels_acc, labels_ign, labels`, there is only `labels` and just some iteration steps are skipped.